cmd/storage-rest-client.go

```diff
@@ -391,6 +391,9 @@ func (client *storageRESTClient) CreateFile(ctx context.Context, origvolume, vol
 }
 
 func (client *storageRESTClient) WriteMetadata(ctx context.Context, origvolume, volume, path string, fi FileInfo) error {
+	ctx, cancel := context.WithTimeout(ctx, globalDriveConfig.GetMaxTimeout())
```

Comment:
isn't ping/pong in the grid code enough ? probably we should fix something in the grid code instead

Reply:
@vadmeste Timeouts are strict timeouts and are forwarded to the remote. So both the caller and the remote

Ping will keep the request alive forever even with no activity - and are only available on streams.
Same with context cancelation - only for streams - cancelling the context will cancel the local, but the remote will have to hit the timeout. Streams will propagate cancelation.

One-shot requests are assumed to be short lived - and have a default deadline of 1 minute, unless something else is set on the context.

Reply:
We cannot rely on pings. Active callers must timeout.